InnerEye/ML/utils/checkpoint_handling.py

```diff
@@ -228,3 +238,179 @@ def get_local_checkpoints_path_or_download(self) -> List[Path]:
         return checkpoint_paths
+
+
+def download_checkpoints_to_temp_folder(run: Optional[Run] = None, workspace: Optional[Workspace] = None) -> Path:
+    """
+    Downloads all files with the outputs/checkpoints prefix of the given run to a temporary folder.
+    In distributed training, the download only happens once per node.
+
+    :param run: If provided, download the files from that run. If omitted, download the files from the current run
+        (taken from RUN_CONTEXT)
+    :param workspace: The AML workspace where the run is located. If omitted, the hi-ml defaults of finding a workspace
+        are used (current workspace when running in AzureML, otherwise expecting a config.json file)
+    :return: The path to which the files were downloaded. The files are located in that folder, without any further
+        subfolders.
+    """
+    run = run or RUN_CONTEXT
+    # Downloads should go to a temporary folder because downloading the files to the checkpoint folder might
+    # cause artifact conflicts later.
+    temp_folder = Path(tempfile.mkdtemp())
+    checkpoint_prefix = f"{DEFAULT_AML_UPLOAD_DIR}/{CHECKPOINT_FOLDER}/"
+    existing_checkpoints = get_run_file_names(run, prefix=checkpoint_prefix)
+    logging.info(f"Number of checkpoints available in AzureML: {len(existing_checkpoints)}")
+    if len(existing_checkpoints) > 0:
+        try:
+            logging.info(f"Downloading checkpoints to {temp_folder}")
+            download_files_from_run_id(run_id=run.id,
+                                       output_folder=temp_folder,
+                                       prefix=checkpoint_prefix,
+                                       workspace=workspace)
+        except Exception as ex:
+            logging.warning(f"Unable to download checkpoints from AzureML. Error: {str(ex)}")
+    # Checkpoint downloads preserve the full folder structure, point the caller directly to the folder where the
+    # checkpoints really are.
+    return temp_folder / DEFAULT_AML_UPLOAD_DIR / CHECKPOINT_FOLDER
+
+
+PathAndEpoch = Tuple[Path, int]
+
+
+def find_recovery_checkpoint_and_epoch(path: Path) -> Optional[PathAndEpoch]:
+    """
+    Looks at all the recovery files, extracts the epoch number for all of them and returns the most recent (latest
+    epoch)
+    checkpoint path along with the corresponding epoch number. If no recovery checkpoint are found, return None.
+    :param path: The folder to start searching in.
+    :return: None if there is no file matching the search pattern, or a Tuple with Path object and integer pointing to
+    recovery checkpoint path and recovery epoch.
+    """
+    available_checkpoints = find_all_recovery_checkpoints(path)
+    if available_checkpoints is None and is_running_in_azure_ml():
+        logging.info("No recovery checkpoints available in the checkpoint folder. Trying to find checkpoints in "
+                     "AzureML from previous runs of this job.")
```

Comment:
I may have misunderstood but I have the impression it will look for checkpoints in the file storage for the current run, rather than a previous run?

Reply:
OK, the wording is not ideal.  What I meant is from a previous re-start of the run. Let me clarify.